client/src/test/java/org/eclipse/hono/client/device/amqp/TelemetrySenderTest.java

```diff
@@ -0,0 +1,167 @@
+
+    /**
+     * Verifies that the message created by {@link TelemetrySender#send(String, byte[], String, Map)} conforms to the
+     * expectations of the AMQP adapter.
+     */
+    @Test
+    public void testSendWithTracing() {
+
+        // GIVEN a TraceableTelemetrySender instance
+        final TraceableTelemetrySender telemetrySender = ((TraceableTelemetrySender) createTelemetrySender());
+
+        // WHEN sending a message using the API
+        final SpanContext spanContext = mock(SpanContext.class);
+        telemetrySender.send(DEVICE_ID, PAYLOAD, CONTENT_TYPE, APPLICATION_PROPERTIES, spanContext);
+
+        // THEN the given SpanContext is used
+        verify(spanBuilder).addReference(any(), eq(spanContext));
+        assertMessageConformsAmqpAdapterSpec(ADDRESS);
+    }
+
+    /**
+     * Verifies that {@link TraceableTelemetrySender#sendAndWaitForOutcome(String, byte[], String, Map, SpanContext)}
+     * uses the given SpanContext.
+     *
+     * @param ctx The test context to use for running asynchronous tests.
+     */
+    @Test
+    public void testSendAndWaitForOutcomeWithTracing(final VertxTestContext ctx) {
+
+        // GIVEN a TraceableTelemetrySender instance
+        final TraceableTelemetrySender telemetrySender = ((TraceableTelemetrySender) createTelemetrySender());
+
+        // WHEN sending a message using the API...
+        final SpanContext spanContext = mock(SpanContext.class);
+        final Future<ProtonDelivery> deliveryFuture = telemetrySender.sendAndWaitForOutcome(DEVICE_ID, PAYLOAD,
+                CONTENT_TYPE, APPLICATION_PROPERTIES, spanContext);
+
+        // ...AND WHEN the disposition is updated by the peer
+        updateDisposition();
+
+        deliveryFuture.setHandler(ctx.succeeding(delivery -> {
+            // THEN the given SpanContext is used
+            ctx.verify(() -> {
+                verify(spanBuilder).addReference(any(), eq(spanContext));
+                assertMessageConformsAmqpAdapterSpec(ADDRESS);
+            });
+            ctx.completeNow();
+        }));
+    }
+
+    /**
+     * Verifies that sending the message waits for the disposition update from the peer.
+     *
+     * @param ctx The test context to use for running asynchronous tests.
+     * @throws InterruptedException if test is interrupted while waiting.
+     */
+    @Test
+    public void testSendAndWaitForOutcomeWaitsForDispositionUpdate(final VertxTestContext ctx)
+            throws InterruptedException {
+
+        // GIVEN a TelemetrySender instance
+        final TelemetrySender telemetrySender = createTelemetrySender();
+
+        // WHEN sending a message using the API
+        final Future<ProtonDelivery> deliveryFuture = telemetrySender.sendAndWaitForOutcome(DEVICE_ID, PAYLOAD,
+                CONTENT_TYPE, APPLICATION_PROPERTIES);
+
+        deliveryFuture.setHandler(ctx.completing());
+
+        // THEN the future waits for the disposition to be updated by the peer
+        Thread.sleep(100L);
```

Comment:
can you please make sure to remove all occurrences of this?

Reply:
Sorry, I missed them.